### PR TITLE
FIX #8155: added type to name map in ingestion pipeline json schema

### LINKED ISSUE #8155: Map task id with pipeline type directly in the JSON Schema
From the convo in https://github.com/open-metadata/OpenMetadata/pull/8145#discussion_r995803318

basically we have some task_id strings in a couple of places. We should centralise the information inside a JSON Schema (wherever it fits the most) and use it everywhere.

### PR DESCRIPTION
**Pull request by finn-mcclusky**
### Describe your changes :

ISSUE #8155

Includes:
- adds type to name map in ingestion pipeline schema

changes made to create a direct mapping between ingestion pipelines type and name

#
### Type of change :

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

**Comment by finn-mcclusky** [comment]
Looking for guidance to make a more acceptable solution as I believe that my mapping schema might not meet the need of the issue nor is the use `getattr(ingestion_pipeline.pipelineTypeToTask, ingestion_pipeline.type)` the best choice.

Any advice would be appreciated.
thank you.

**Comment by TeddyCr** [comment]
cc @pmbrull

**Comment by harshach** [comment]
@pmbrull what you think of these changes? should we merge it in?

**Comment by pmbrull** [comment]
oooh apologies missed this one, yeah the approach LGTM!!

File: openmetadata-spec/src/main/resources/json/schema/entity/services/ingestionPipelines/ingestionPipeline.json [hunk 1]
```diff
@@ -13,6 +13,40 @@
...
+        },
+        "lineage": {
+          "type": "string",
+          "default": "lineage_task"
+        },
+        "usage": {
+          "type": "string",
+          "default": "usage_task"
+        },
+        "TestSuite": {
```
Comment (nahuelverdugo) [inline openmetadata-spec/src/main/resources/json/schema/entity/services/ingestionPipelines/ingestionPipeline.json:37]:
```suggestion
        "testSuite": {
```

File: openmetadata-spec/src/main/resources/json/schema/entity/services/ingestionPipelines/ingestionPipeline.json [hunk 2]
```diff
@@ -13,6 +13,40 @@
...
+          "type": "string",
+          "default": "test_suite_task"
+        }
+      },
+      "required": [
+        "metadata",
+        "profiler",
+        "lineage",
+        "usage",
+        "TestSuite"
```
Comment (nahuelverdugo) [inline openmetadata-spec/src/main/resources/json/schema/entity/services/ingestionPipelines/ingestionPipeline.json:47]:
```suggestion
        "testSuite"
```

**Review by nahuelverdugo — COMMENTED** [review]
@pmbrull I think we can have `pipelineTypeToTask` outside the `IngestionPipeline` object ad have a different name like `pipelineTypeTaskMap`, what do you think? I have also added some comment to have a more consistent code.

**Comment by pmbrull** [comment]
> @pmbrull I think we can have `pipelineTypeToTask` outside the `IngestionPipeline` object and have a different name like `pipelineTypeTaskMap`, what do you think? I have also added some comments to have a more consistent code.

Thanks, @nahuelverdugo, that's a good point. We can indeed separate this and keep the Ingestion Pipeline a bit cleaner.

**Comment by pmbrull** [comment]
closing this out. We'll need to think a bit more on that

**Pull request closed without merging**